ast/src/parsed/display.rs

```diff
@@ -587,6 +595,76 @@ fn format_list<L: IntoIterator<Item = I>, I: Display>(list: L) -> String {
     format!("{}", list.into_iter().format(", "))
 }
 
+impl<E: Display> Expression<E> {
+    pub fn precedence(&self) -> Option<ExpressionPrecedence> {
+        match self {
+            Expression::UnaryOperation(op, _) => Some(op.precedence()),
+            Expression::BinaryOperation(_, op, _) => Some(op.precedence()),
+            _ => None,
```

Comment:
Shouldn't you define precedence also for array and lambda expression?

Reply:
It's not required with our configuration in my view. They have more precedence than others and are postfix.